packages/discord.js/src/managers/GuildMemberManager.js

```diff
@@ -441,6 +441,37 @@ class GuildMemberManager extends CachedManager {
     return this.guild.bans.remove(user, reason);
   }
 
+  /**
+   * The options for {@link GuildMemberManager#addRole} and {@link GuildMemberManager#removeRole}.
+   * @typedef {Object} RoleModifyOptions
+   * @property {RoleResolvable} role The role to add or remove
```

Comment:
Should the role be in the options? It's required variable that can be in a separate parameter.

Reply:
It's really for consistency with other manager methods because typically, we would add helpers to the class that the manager is handling.

For example, if we were to add this helper to the `GuildMember` class, it would be called `GuildMember#addRole()`. The first parameter would be, of course, `RoleModifyOptions` which contains the role and the reason. Just in case it ever happens, we won't have that problem in the future.

Reply:
But the rule with djs is usually that required parameters come alone and options go in an object, So I agree with Syjalo

Reply:
A change in v14 is that `name` goes in the options of methods now, despite it being required

Reply:
> A change in v14 is that `name` goes in the options of methods now, despite it being required

But it's because the name is also part of the data

Reply:
Yeah here the role and the reason are sent separately while in that example the name goes with the rest of the data

Reply:
I'm going to change the parameters to `(user, role, reason)` then, and remove the `RoleModifyOptions`

Reply:
I have not found any manager method in which the required parameter is included in the options, except when the parameter is part of the data or it's the fetch method, which can return a single structure or  a collection of structures.
[GuildChannelManager#setPosition()](https://discord.js.org/#/docs/discord.js/main/class/GuildChannelManager?scrollTo=setPosition) has 2 required parameters and `position` is not in the options.